### PR TITLE
Simplify flat packed-document attention mask

### PR DESCRIPTION
**Pull request by drisspg**
## Agent note

This is a focused draft against the head of #4121. After folding the batch dimension into the token
dimension, each packed token's position directly encodes its distance from the start of its document.
Use `q_idx - positions[q_idx]` as the document start instead of constructing document IDs and a
padded offset table.

This retains the specialized causal packed-document path, which remains faster than composing the
generic document-ID mask with a causal mask, while removing its preprocessing and indirect lookup.

## Performance

Measured on an NVIDIA B200 with PyTorch `2.15.0.dev20260818+cu132`, bf16 Q/K/V shaped
`[1, 16, 8192, 128]`, document length 1024, and 100 CUDA-graph replays per result. Values are the
median forward + backward latency across three interleaved rounds.

| Backend | Generic document IDs | Existing offsets | Direct positions |
|---|---:|---:|---:|
| FlexFlash | 417.520 us | 387.008 us | 364.144 us |
| Triton | 847.376 us | 636.640 us | 609.776 us |

## Test Plan

```bash
PYTHONPATH=$PWD ~/.venvs/nightly/bin/pytest tests/unit_tests/test_chat_dataset.py::TestDocumentMaskBlocksCrossDocAttention::test_packed_document_mask_composes_with_causal_mask tests/unit_tests/test_flex_attention.py -q
```

### CODE REVIEW

File: torchtitan/models/common/attention.py [hunk 1]
```diff
@@ -450,34 +449,18 @@ def get_efficient_causal_mask_mod_for_packed_document(
...
-            document_starts, document_id, torch.full_like(document_id, seq_len)
-        ).to(torch.int64),
-        torch.where(document_starts, token_idx, torch.full_like(token_idx, seq_len)),
-    )
 
     def packed_document_mask(
         b: torch.Tensor, h: torch.Tensor, q_idx: torch.Tensor, kv_idx: torch.Tensor
     ) -> torch.Tensor:
-        return kv_idx >= offsets[document_id[q_idx]]
+        return kv_idx >= q_idx - positions[q_idx]
```
Comment (fegin) [inline torchtitan/models/common/attention.py:463]: Pretty clean and neat now!

**Review by fegin — APPROVED** [review]

**Comment by drisspg** [comment]
I shall wait for https://github.com/pytorch/torchtitan/pull/4121 to land and fix lint :)

File: torchtitan/models/common/attention.py
Comment (tianyu-l) [inline torchtitan/models/common/attention.py:418]: do we need to keep this? Can we remove and let the efficient one have this name
